iac-extensions/arm/src/main/java/org/sonar/iac/arm/checkdsl/ContextualResource.java

```diff
@@ -66,6 +71,27 @@ protected HasTextRange toHighlight() {
   }
 
   public ContextualResource childResourceBy(String type, Predicate<ResourceDeclaration> predicate) {
+    return Optional.ofNullable(nestedChildResourceBy(type, predicate))
+      .filter(ContextualResource::isPresent)
+      .orElseGet(() -> childResourceOutsideOfThisBy(type, predicate));
+  }
+
+  private ContextualResource childResourceOutsideOfThisBy(String type, Predicate<ResourceDeclaration> predicate) {
+    var topLevelResources = Optional.ofNullable(tree)
+      .map(ResourceDeclaration::parent)
```

Comment:
This might need a bit more attention, because now I'm not diving deep into the tree and only checking one level of depth. When jumping to the root node, a full-tree search should be performed to catch the case you described.